src/test/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSparkUnitTest.java

```diff
@@ -0,0 +1,95 @@
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceWindowFunctions;
+import org.broadinstitute.hellbender.engine.spark.SparkContextFactory;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Unit tests for FindBadGenomicKmersSpark
+ */
+public class FindBadGenomicKmersSparkUnitTest extends BaseTest {
+
+    private static final int KMER_SIZE = SVConstants.KMER_SIZE;
+    private static final String REFERENCE_FILE_NAME = hg19MiniReference;
+
+    @Test(groups = "spark")
+    public void badKmersTest() throws IOException {
+        final File refSequenceFile = File.createTempFile("genomic_sequence", "txt");
+        refSequenceFile.deleteOnExit();
+
+        final String polyA = StringUtils.repeat('A', KMER_SIZE);
+        final String polyC = StringUtils.repeat('C', KMER_SIZE);
+        final String polyT = StringUtils.repeat('T', KMER_SIZE);
+
+        try ( PrintWriter writer = new PrintWriter(refSequenceFile) ) {
+            long nTimes = FindBadGenomicKmersSpark.MAX_KMER_FREQ;
+            // write polyA and polyC the max number of times possible while evading the trigger
+            while ( nTimes-- > 0L ) {
+                writer.println(polyA);
+                writer.println(polyC);
+            }
+            // tip polyA over the edge by writing its reverse complement
+            writer.println(polyT);
+        }
+
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        final List<SVKmer> badKmers = FindBadGenomicKmersSpark.processRefFile(ctx, refSequenceFile);
+
+        // should have just one bad kmer:  polyA
+        Assert.assertEquals(badKmers.size(), 1);
+        Assert.assertEquals(badKmers.get(0).toString(KMER_SIZE), polyA);
+
+        if ( !refSequenceFile.delete() )
+            throw new GATKException("Unable to delete file "+refSequenceFile);
+    }
+
+    @Test(groups = "spark")
+    public void miniRefTest() throws IOException {
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        final ReferenceMultiSource ref = new ReferenceMultiSource((PipelineOptions)null,
+                REFERENCE_FILE_NAME,
+                ReferenceWindowFunctions.IDENTITY_FUNCTION);
+        final SAMSequenceDictionary dict = ref.getReferenceSequenceDictionary(null);
+        if ( dict == null ) throw new GATKException("No reference dictionary available.");
+
+        final HashMap<SVKmer,Long> kmerMap = new HashMap<>();
+        for ( final SAMSequenceRecord rec : dict.getSequences() ) {
+            final String seqName = rec.getSequenceName();
```

Comment:
I'd tend towards inlining things like `seqName` and `seqLength` since they're just simple getters and aren't reused.  I wouldn't inline `bases` since it's a bit more complicated of a call and I think it's clearer the way you have it here.

No need to change anything, this is just me rambling about personal preferences.